cdap-ui/app/cdap/components/AbstractWidget/DLPCustomWidget/DLPRow.tsx

```diff
@@ -0,0 +1,398 @@
+  public static defaultProps = {
+    transforms: [],
+    filters: [],
+    extraConfig: '',
+    expanded: false,
+  };
+
+  public state = {
+    fields: '',
+    transform: '',
+    filters: '',
+    transformProperties: {},
+    expanded: false,
+    transformObj: null,
+  };
+
+  public componentDidMount() {
+    try {
+      const jsonString = this.props.value;
+      const jsonObj = JSON.parse(jsonString);
+      jsonObj.transformObj =
+        this.props.transforms.filter((t) => t.name === jsonObj.transform)[0] || null;
+      this.configurationGroupPluginProperties = parseTransformOptions(jsonObj.transformObj);
+      this.configurationGroupWidgetJson =
+        jsonObj.transformObj !== null
+          ? {
+              'configuration-groups': [
+                {
+                  label: jsonObj.transformObj.label + ' properties',
+                  properties: jsonObj.transformObj.options,
+                },
+              ],
+              filters: jsonObj.transformObj.filters, // Passing ConfigurationGroup filters to allow for dynamic showing/hiding of properties
+            }
+          : {};
+      this.setState({ ...this.state, ...jsonObj });
+    } catch (error) {
+      this.setState({
+        fields: '',
+        transform: '',
+        filters: '',
+        transformProperties: {},
+      });
+    }
+  }
+
+  // Handles onChange for MultiSelect components (filters and fields selectors)
+  private handleChangeMultiSelect = (type: StateKeys, e: string) => {
+    if (type === 'filters') {
+      const noneWasSelected: boolean = this.state.filters.includes('NONE');
+      const noneIsSelected: boolean = e.includes('NONE');
+
+      // Ensuring user can't select 'NONE' along with other filters
+      // TODO: Use another component for the filters selector (https://issues.cask.co/browse/CDAP-16124)
+      if (noneWasSelected && noneIsSelected) {
+        e = e.replace('NONE,', '');
+      } else if (!noneWasSelected && noneIsSelected) {
+        e = 'NONE';
+      }
+    }
+
+    this.handleChange({ [type]: e });
+  };
+
+  // Handles onChange for Select components (transform selector)
+  private handleChangeTransformSelect = (type: StateKeys, e) => {
+    const value = e.target.value;
+    if (value === this.state.transform) {
+      return;
+    }
+    const newValues = { [type]: value };
+
+    const transformProps = {};
+    this.props.transforms
+      .filter((transform) => transform.name === value)[0]
+      .options.forEach((element) => {
+        transformProps[element.name] = element['widget-attributes'].default || '';
+      });
+    newValues.transformProperties = transformProps;
+    newValues.expanded = Object.keys(transformProps).length > 0;
+    newValues.transformObj = this.props.transforms.filter((t) => t.name === value)[0] || null;
+    this.configurationGroupPluginProperties = parseTransformOptions(newValues.transformObj);
+    this.configurationGroupWidgetJson =
+      newValues.transformObj !== null
```

Comment:
nit: This is hard to read. Can we make this a simple `if...else` condition?

Reply:
done